app/org/sagebionetworks/bridge/services/backfill/UploadValidationBackfill.java

```diff
@@ -65,48 +81,56 @@ public final void setUploadValidationService(UploadValidationService uploadValid
-            try {
-                Thread.sleep(1000);
-            } catch (InterruptedException ex) {
-                logger.error("Interrupted while sleeping: " + ex.getMessage(), ex);
-            }
-
-            try {
-                // If it's already succeeded, don't backfill it
-                Upload oneUpload = uploadDao.getUpload(uploadId);
-                if (oneUpload.getStatus() == UploadStatus.SUCCEEDED) {
-                    logger.info("Upload " + uploadId + " already succeeded, skipping backfill");
-                    continue;
+        // get file of list of record IDs
+        S3Object recordIdFile = s3Client.getObject(RECORD_ID_BUCKET, RECORD_ID_FILENAME);
+        try (BufferedReader recordIdReader = new BufferedReader(new InputStreamReader(recordIdFile.getObjectContent(),
+                Charsets.UTF_8))) {
+            // query and iterate over record IDs
+            String oneRecordId;
+            while ((oneRecordId = recordIdReader.readLine()) != null) {
+                // rate limit so we down starve threads or brown out DDB
+                try {
+                    Thread.sleep(1000);
+                } catch (InterruptedException ex) {
+                    logger.error("Interrupted while sleeping: " + ex.getMessage(), ex);
                 }
 
-                // Call uploadComplete() to reset the upload status and uploadDate. This will make the upload
-                // eligible for validation again.
-                uploadDao.uploadComplete(oneUpload);
-
-                // Get study ID from health code. Upload validation needs this.
-                String studyId = healthCodeDao.getStudyIdentifier(oneUpload.getHealthCode());
-                StudyIdentifier studyIdentifier = new StudyIdentifierImpl(studyId);
-
-                // Kick off upload validation.
-                uploadValidationService.validateUpload(studyIdentifier, oneUpload);
-
-                recordMessage(task, callback, "Backfilled upload ID " + uploadId);
-                logger.info("Backfilled upload ID " + uploadId);
-            } catch (RuntimeException ex) {
-                // Ensure that errors won't fail the entire backfill. Log an error and move on.
-                String errMsg = "Error backfilling upload ID " + uploadId + ": " + ex.getMessage();
-                logger.error(errMsg, ex);
-                recordMessage(task, callback, errMsg);
+                try {
+                    // get upload ID for record ID
+                    HealthDataRecord record = healthDataDao.getRecordById(oneRecordId);
+                    String uploadId = record.getUploadId();
+
+                    // Call uploadComplete() to reset the upload status and uploadDate. This will make the upload
+                    // eligible for validation again.
+                    Upload oneUpload = uploadDao.getUpload(uploadId);
+                    uploadDao.uploadComplete(oneUpload);
```

Comment:
Would resetting the upload date be Ok? I understand it is needed for automatic redriving. But if we reset it,  if we ever want to find the original upload date, it would be in the S3 object's metadata?